src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,152 @@
+            String result = tasks.doneTask(markNumber);
+            storage.saveTasks(this.tasks.getTaskList());
+            return result;
+
+        }  else if (commandType == Commands.LIST) {
+
+            String[] extractedData = parser.extractData(isLoaded, tokens);
+            String content = extractedData[0];
+            DateTimeHelper dtHelper = DateTimeHelper.processDateTime(content);
+            if (dtHelper != null) {
+                LocalDate deadline = dtHelper.getDate();
+                return tasks.filteredTaskList(deadline);
+            } else {
+                throw new DukeException("Wrong format\n"
+                        + " Your date and time(optional) should be in this format:\n"
+                        + "yyyy-mm-dd HHmm\ne.g: 2019-10-15 1800 or 2019-10-15");
+            }
+
+        } else if (commandType == Commands.FIND) {
+            return tasks.findTasks(tokens);
+        } else if (commandType == Commands.BYE || commandType == Commands.HELP) {
+            throw new DukeException("Wrong format of command " + "'" + commandType.getAction() + "'");
+        }
+
+        if (!parsedTask.getContent().equals("")) {
+            this.tasks.addTask(parsedTask);
+        }
+        storage.saveTasks(this.tasks.getTaskList());
+        return parsedTask.getUiOutput();
+    }
+
+    /**
+     * Strips trailing whitespaces and tokenizes each command before further processing
+     * @param command
+     * @param isLoaded
+     * @return message to user
+     * @throws DukeException
+     */
+    protected String processedCommand(String command, boolean isLoaded) throws DukeException {
+        command = command.strip();
+        if (command.equals("")) return "";
+        String[] tokens = command.split(" ");
+        try {
+            return furtherProcessing(Commands.valueOf(tokens[0].toUpperCase()), tokens, isLoaded);
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("OOPS!!! CAN YOU PLEASE TYPE SOMETHING MEANINGFUL?");
+        }
+    }
+
+    /**
+     * Reads input and return output to the user
+     */
+    protected String init() {
+        String result = "";
+        result += Ui.printDialog("Hello! I'm Elon Musk. Type 'help' if you know nothing HAHAHA\n"
+                + "Your tasks will be saved at /data\nWhat can WE do for you?");
+        ArrayList<String> savedTasks = storage.loadSavedTasks();
+        if (savedTasks.size() > 0 && savedTasks.get(0).equals("000")) {
+            result += Ui.printDialog("This is the first time you use Duke!");
+        } else {
+            try {
+                for (String task : savedTasks) {
+                    processedCommand(task, true);
+                }
+            } catch (DukeException e) {
+                result += Ui.printDialog("Something wrong happened while loading saved tasks");
+            }
+        }
+        return result;
+    }
+    public String getResponse(String content) {
```

Comment:
Would leaving a description for this method allow others to better understand what this method does?

Reply:
Yeah I assumed that the name was descriptive enough. Thanks for the feedback, I will add more descriptions.